CorsixTH/Lua/app.lua

```diff
@@ -1263,13 +1263,18 @@ function App:checkInstallFolder()
       " as said files are required for graphics and sounds."
   if not status then
     -- Table of predictable places. First three are platform independent,
-    -- then macOS GOG bundle, then linux Filesystem Hierarchy Standard, then Windows
-    local possible_locations = { os.getenv( "HOME" ), os.getenv( "HOME" ) .. pathsep ..  "Documents",
+    -- then macOS app and its parent folder, GOG bundle,
+    -- then linux Filesystem Hierarchy Standard, then Windows Program Files
+    -- root_dir is the macOS app base directory named CorsixTH.app
+    local app_parent_dir = debug.getinfo(1).short_src:match("(.*)/Contents/.")
+    local user_dir = os.getenv("HOME") or os.getenv("HOMEPATH") or ""
+    local possible_locations = { user_dir, user_dir .. pathsep ..  "Documents",
       select(1, corsixth.require("config_finder")):match("(.*[/\\])"):sub(1, -2),
+      app_parent_dir, app_parent_dir and app_parent_dir:sub(1,-14),
```

Comment:
14 is a magic number. Please clarify.

Reply:
Fixed to be more like the comment describes.